library/ssl_tls.c

```diff
@@ -2024,8 +2024,8 @@ STATIC int ssl_decrypt_buf( mbedtls_ssl_context *ssl,
                                     rec->data_len );
             mbedtls_md_hmac_finish( &transform->md_ctx_dec, mac_expect );
             /* Call mbedtls_md_process at least once due to cache attacks */
-            for( j = 0; j < extra_run + 1; j++ )
-                mbedtls_md_process( &transform->md_ctx_dec, data );
+            //for( j = 0; j < extra_run + 1; j++ )
```

Comment:
Suggested fix:
```
/* top of the block */
unsigned char dummy[MBEDTLS_MD_MAX_BLOCK_SIZE];
memset( dummy, 0, sizeof( dummy ) );
/* here */
for( j = 0; j < extra_run + 1; j++ )
    mbedtls_md_process( &transform->md_ctx_dec, dummy );
```
where `MD_MAX_BLOCK_SIZE` needs to be defined in `md.h` as 128 if `SHA512_C` is defined and 64 otherwise.

Reply:
@mpg Thank you, I will follow that suggestion.